morphia/src/main/java/org/mongodb/morphia/AdvancedDatastore.java

```diff
@@ -247,6 +254,16 @@
     void setDecoderFact(DBDecoderFactory fact);
 
     /**
+     * Returns the {@see MongoCollection} for the given name
+     *
+     * @param <T>   the type of the MongoCollection
+     * @param name  the name of the collection
+     * @param clazz the type of the collection
+     * @return the collection
+     */
+    <T> MongoCollection<T> getMongoCollection(String name, Class<T> clazz);
```

Comment:
It's no less an abstraction leak than the existing getCollection() methods.  It can probably hidden in *this* case but the plan is to move everything to `MongoCollection` eventually and this will only open back up.